src/components/CmsZone.vue

```diff
@@ -369,6 +397,28 @@ export default class CmsZone extends Vue {
   get isInspectOverlayEnabled(): boolean {
     return this.$cms.isInspectOverlayEnabled;
   }
+
+  private updateSearchQuery(query: string): void {
+    // Doing this rather than a straight property set so that the @Watch on extra fires
+    this.extra = { ...this.extra, q: query };
+    if (query) {
+      pluginOptions.trackAnalytics('search cms conent', { 'search query': query });
+    }
+  }
+
+  private updateSelectedCategory(filterCategory: string): void {
+    if (filterCategory !== 'All') {
+      this.extra = { ...this.extra, category: filterCategory };
+    } else {
+      // Destructuring assignment
+      const { category, ...extrasWithoutCategory } = this.extra;
+      this.extra = { ...extrasWithoutCategory };
```

Comment:
This is cool - didn't know this worked!

Reply:
Wait... noticed in demo today - how do the following two lines differ?
```javascript
this.extra = { ...extrasWithoutCategory };
this.extra = extrasWithoutCategory;
```

Reply:
good question had to do some reading to make sure i understood the diff but i think i've got it. so the way that destructuring works is that it creates a new object and assigns properties to that object by reference. so the properties of extrasWithoutCategory still just point to the properties of extra. on the other hand, using the spread operator takes those properties and creates brand new points in memory for the properties of the new object. this is only important because there is a [known issue ](https://vuejs.org/v2/guide/reactivity.html#For-Objects) with Vue Change Detection for Objects where it can't detect property addition or removal from watched objects.